### PR TITLE
Add details for shipment type Delivery

### PR DESCRIPTION
**Pull request by nishantjinsi**

### CODE REVIEW

**Review by bdbelevate — APPROVED** [review]

File: src/models/enums/schedule_type.rs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+//! Model for Schedule type enum
+
+use serde::{Deserialize, Serialize};
+
+/// Indicates the fulfillment delivery schedule type. If SCHEDULED, then deliver_at is required.
+/// If ASAP, then prep_time_duration is required. The default is SCHEDULED.
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
+pub enum ScheduleType {
```
Comment (mharkins-cosm) [inline src/models/enums/schedule_type.rs:9]: `OrderFulfillmentDeliveryDetailsScheduleType`

File: src/models/enums/schedule_type.rs [hunk 2]
```diff
@@ -0,0 +1,14 @@
+//! Model for Schedule type enum
+
+use serde::{Deserialize, Serialize};
+
+/// Indicates the fulfillment delivery schedule type. If SCHEDULED, then deliver_at is required.
```
Comment (mharkins-cosm) [inline src/models/enums/schedule_type.rs:5]: I've just been using the documentation from the Square docs for the type:
```rust
/// The schedule type of the delivery fulfillment.
```
See https://developer.squareup.com/reference/square_2022-12-14/enums/OrderFulfillmentDeliveryDetailsScheduleType

**Review by mharkins-cosm — CHANGES_REQUESTED** [review]
Just for consistency's sake and for avoiding name clashes with future "schedule types", please see comments.

File: src/models/enums/schedule_type.rs (hunk 1 above)
Comment (nishantjinsi) [reply src/models/enums/schedule_type.rs]: 👍

File: src/models/enums/order_fulfillment_delivery_details_schedule_type.rs [hunk 3]
```diff
@@ -0,0 +1,14 @@
+//! Model for order_fulfillment_delivery_details_schedule_type enum
```
Comment (mharkins-cosm) [inline src/models/enums/order_fulfillment_delivery_details_schedule_type.rs:1]: `//! Model for OrderFulfillmentDeliveryDetailsScheduleType enum`

File: src/models/enums/order_fulfillment_delivery_details_schedule_type.rs [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+use serde::{Deserialize, Serialize};
+
+/// Indicates the fulfillment delivery schedule type. If SCHEDULED, then deliver_at is required.
+/// If ASAP, then prep_time_duration is required. The default is SCHEDULED.
```
Comment (mharkins-cosm) [inline src/models/enums/order_fulfillment_delivery_details_schedule_type.rs:6]: `/// The schedule type of the delivery fulfillment.`

**Review by mharkins-cosm — APPROVED** [review]

**Pull request merged**